File: go/signal124/signal124_test.go
package signal124

import "testing"

var samples = map[string]rune{
	"Hell-o World": "tØàù;Θf£&0ì8E(Å",
	"John Doe": "J#c+§ΦèCØå",
	"signal01": "CΩÄ+'6¿£/ù",
}

func TestEncode(t *testing.T) {
	for k, v := range samples {
		if val := EncodeToString([]byte(k)); val != v {
			t.Errorf("Incorrect encoding of `%s` got `%s`", k, val)
		} else {
			t.Logf("Correct encoding of `%s` got `%s`", k, val)
		}
	}
}

func TestDecode(t *testing.T) {
	for k, v := range samples {
		if val := DecodeToString([]byte(v)); val != k {
			t.Errorf("Incorrect decoding of `%s` got `%s`", v, val)
		} else {
			t.Logf("Correct decoding of `%s` got `%s`", v, val)
		}
	}
}

File: php/signal124/signal124_test.php
<?php
header("Content-Type: text/plain");
include "../signal124.php";

$samples = Array(
	"Hell-o World" => "tØàù;Θf£&0ì8E(Å",
	"John Doe" => "J#c+§ΦèCØå",
	"signal01" => "CΩÄ+'6¿£/ù"
);

function TestEncode() {
	global $samples;
	foreach ($samples as $dec => $enc) {
		$encoded = signal124Encode($dec);
		if($encoded != $enc) {
			printf("Incorrect encoding of `%s` got `%s`\n", $enc, $encoded);
		} else {
			printf("Correct encoding of `%s` got `%s`\n", $enc, $encoded);
		}
	}
}

function TestDecode() {
	global $samples;
	foreach ($samples as $dec => $enc) {
		$decoded = signal124Decode($enc);
		if($decoded != $dec) {
			printf("Incorrect decoding of `%s` got `%s`\n", $dec, $decoded);
		} else {
			printf("Correct decoding of `%s` got `%s`\n", $dec, $decoded);
		}
	}
}

TestEncode();
print("\n");
TestDecode();
?>